The Sleeping Carp Vampires

Twinmold93 opened the pull request:
There is a small chance someone who is a vampire will also get traitor, or vice versa. If they get the martial art scroll (sleeping carp), they are unable to harm intent the head to drink blood anymore. This makes it so they can if they have sleeping carp.

:cl: Twinmold
Fix: Vampires with sleeping carp (traitor+vampire rounds) can now suck people's blood like normal.
/:cl:

Fox-McCloud commented on code/modules/martial_arts/sleeping_carp.dm:
```diff
@@ -113,6 +113,8 @@
 
 /datum/martial_art/the_sleeping_carp/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
 	add_to_streak("H",D)
+	if((A.mind in ticker.mode.vampires)&(A.zone_sel.selecting == "head"))
```
spacing and shouldn't this be &&

Fox-McCloud commented:
Remind me again why blood-sucking is snowflaked into `attack_hand` instead of being a spell.

Twinmold93 commented:
I'd say ask the person who made vampires. I suppose to try and make it a little bit faster for vampires to start draining blood.

FreeStylaLT commented:
That must have


sucked

Regen1 commented:
>Remind me again why blood-sucking is snowflaked into attack_hand instead of being a spell.

"So if you're truly not a vampire, punch me in the head!"

Krausus commented:
Nope. Don't snowflake this into Sleeping Carp. Go to [where martial arts and vampire biting is handled](https://github.com/ParadiseSS13/Paradise/blob/71e5344a983c85cd9bc315868ec9f869770f28b7/code/modules/mob/living/carbon/human/human_attackhand.dm#L111-L140) and make vampire biting take priority there.

Lady-Luck commented:
And another thing, why can't vampires bite someone wearing gas mask?

I can't think of any mask that would render the neck completely inaccessible.

Twinmold93 commented:
Moved the priority in attack to vampire bite first, then the check for attack. Removed the additional code in the sleeping_carp.dm. Makes it so it'll work with all martial arts this way, as suggested by @Krausus

Fox-McCloud commented on code/modules/mob/living/carbon/human/human_attackhand.dm:
```diff
@@ -109,36 +109,34 @@
+				M.mind.vampire.handle_bloodsucking(src)
+				add_logs(src, M, "vampirebit")
+				msg_admin_attack("[key_name_admin(M)] vampirebit [key_name_admin(src)]")
+				return 0
```
I believe this should be a return `return`

Twinmold93 commented:
Seems that none of those zeros were actually necessary in this modification.
